src/instructions/handlers_storage.zig

```diff
@@ -109,6 +109,11 @@ pub fn Handlers(comptime FrameType: type) type {
                 };
             }
 
+            // EIP-214: Check static context - no storage writes in static calls
+            if (evm.is_static_context()) {
+                return Error.WriteProtection;
+            }
+            
```

Comment:
can't accept this pr because we added branching to every storage check that is unnecessary. This  branching here will hurt performance of amajor path. We have an existing solution for this called StaticDatabase